std/json.d

```diff
@@ -243,32 +243,68 @@ struct JSONValue
     /// Value getter/setter for $(D JSON_TYPE.OBJECT).
     /// Throws: $(D JSONException) for read access if $(D type) is not
     /// $(D JSON_TYPE.OBJECT).
-    @property ref inout(JSONValue[string]) object() inout pure
+    /* Note: this is @system because of the following pattern:
+       ---
+       auto a = &(json.object());
```

Comment:
As an aside, much like we've disallowed taking the address of ref parameters in `@safe` we must disallow taking the address of ref return values in `@safe`. See e.g. [this issue report](https://issues.dlang.org/show_bug.cgi?id=15191). Only tangentially related to this getter as it still wouldn't have a memory safe interface (returning address to union member is bad as long as there are properties that can overwrite it with a different type).